core/Archive/DataTableFactory.php

```diff
@@ -157,21 +161,62 @@ public function useSubtable($idSubtable)
      */
     public function make($index, $resultIndices)
     {
+        $keyMetadata = $this->getDefaultMetadata();
+
         if (empty($resultIndices)) {
             // for numeric data, if there's no index (and thus only 1 site & period in the query),
             // we want to display every queried metric name
             if (empty($index)
-                && $this->dataType == 'numeric'
+                && $this->isNumericDataType()
             ) {
                 $index = $this->defaultRow;
             }
 
-            $dataTable = $this->createDataTable($index, $keyMetadata = array());
+            $dataTable = $this->createDataTable($index, $keyMetadata);
         } else {
-            $dataTable = $this->createDataTableMapFromIndex($index, $resultIndices, $keyMetadata = array());
+            $dataTable = $this->createDataTableMapFromIndex($index, $resultIndices, $keyMetadata);
         }
 
-        $this->transformMetadata($dataTable);
+        return $dataTable;
+    }
+
+    /**
+     * Creates a merged DataTable|Map instance using an index of archive data similar to {@link make()}.
+     *
+     * Whereas {@link make()} creates a Map for each result index (period and|or site), this will only create a Map
+     * for a period result index and move all site related indices into one dataTable. This is the same as doing
+     * `$dataTableFactory->make()->mergeChildren()` just much faster. It is mainly useful for reports across many sites
+     * eg `MultiSites.getAll`. Was done as part of https://github.com/piwik/piwik/issues/6809
+     *
+     * @param array $index @see DataCollection
+     * @param array $resultIndices an array mapping metadata names with pretty metadata labels.
+     *
+     * @return DataTable|DataTable\Map
+     * @throws \Exception
+     */
+    public function makeMerged($index, $resultIndices)
+    {
```

Comment:
If you don't want to make this addition to core generic (and thus not tied to a specific use case in a single plugin), can you do one of the following things:

1) _(quickest)_ Mark Archive::getDataTableNumericMerged or whatever it is called as `@internal` and create an issue about removing it and archiving MultiSites.getAll. Please leave a comment that says the Archive & DataTableFactory methods should be dealt w/ some other way if this new ticket is considered invalid and closed w/o being implemented.
2) Move the logic to the MultiSites plugin. If another plugin needs something like it, then it can be moved to core.